Update HIP definition (#2134)
ROCm/rocm-systems: projects/hip/docs/what_is_hip.rst

.. meta::
  :description: This chapter provides an introduction to the HIP API.
  :keywords: AMD, ROCm, HIP, C++ language extensions

.. _intro-to-hip:

*******************************************************************************
What is HIP?
*******************************************************************************

The Heterogeneous-computing Interface for Portability (HIP) API is a C++ runtime
API and kernel language that lets developers create portable applications
running in heterogeneous systems, using CPUs and AMD GPUs from a single source

> code.

Reviewer: ```suggestion
The Heterogeneous-computing Interface for Portability (HIP) API is a C++ runtime
API and kernel language that lets developers create portable applications
that run on heterogeneous systems, using CPUs and AMD GPUs from a single source
code base.
```

Reviewer: Why is AMD GPUs but the CPUs are not only AMD?

Reviewer: Why doesn't the intro mention ROCm?

Reviewer 2: CPUs can be from anyone, the GPUs are ours. 

We could mention that HIP is a part of the ROCm platform or ecosystem or something along those lines, as shown in the ROCm stack.

Reviewer 3: Fixed, let me know if this is better

.. figure:: data/what_is_hip/hip.svg
    :alt: HIP in an application.
    :align: center

* HIP is a thin API with little or no performance impact over coding directly

>   in AMD :doc:`ROCm <rocm:what-is-rocm>`.

Reviewer: What does coding in AMD mean? AMD is a company
We should be more specific.

Reviewer 2: This reads as "coding directly in AMD ROCm."

Reviewer 3: The text will read as AMD ROCm in the preview

> * HIP enables coding in a single-source C++ programming language including

Reviewer: ```suggestion
* HIP enables coding in a single-source C++ programming language, including
```

Reviewer 3: Fixed

  features such as templates, C++11 lambdas, classes, namespaces, and more.

> * Developers can specialize for the platform to tune for performance or handle

Reviewer: "Specialize for the platform" sounds odd>
what is this trying to say.

Reviewer 2: Maybe just go with "Developers can tune for..."

Reviewer 3: Fixed, let me know if this is better

  tricky cases.

ROCm offers compilers (``clang``, ``hipcc``), code profilers (``rocprofv3``),
debugging tools (``rocgdb``), libraries and HIP with the runtime
API and kernel language, to create heterogeneous applications running on both
CPUs and GPUs. ROCm provides marshalling libraries like
:doc:`hipFFT <hipfft:index>` or :doc:`hipBLAS <hipblas:index>` that act as a
thin programming layer over AMD ROCm and offer API compatiblity with CUDA
equivalent library. These libraries offer pointer-based memory interfaces and

> are easily integrated into your applications.

Reviewer: ```suggestion
thin programming layer over AMD ROCm and offer API compatibility with the equivalent NVIDIA CUDA
libraries. These libraries provide pointer-based memory interfaces and
can be easily integrated into your applications.
```

Reviewer 3: Fixed

HIP supports the ability to build and run on either AMD GPUs or NVIDIA GPUs.

> GPU Programmers familiar with NVIDIA CUDA or OpenCL will find the HIP API

Reviewer: ```suggestion
HIP supports the building and running on both AMD and NVIDIA GPUs.
GPU Programmers familiar with CUDA or OpenCL will find the HIP API
```

Reviewer 3: Fixed

What is HIP?
*******************************************************************************

The Heterogeneous-computing Interface for Portability (HIP) API is a C++ runtime
API and kernel language that lets developers create portable applications
running in heterogeneous systems, using CPUs and AMD GPUs from a single source
code.

.. figure:: data/what_is_hip/hip.svg
    :alt: HIP in an application.
    :align: center

* HIP is a thin API with little or no performance impact over coding directly
  in AMD :doc:`ROCm <rocm:what-is-rocm>`.

* HIP enables coding in a single-source C++ programming language including
  features such as templates, C++11 lambdas, classes, namespaces, and more.

* Developers can specialize for the platform to tune for performance or handle
  tricky cases.

ROCm offers compilers (``clang``, ``hipcc``), code profilers (``rocprof``,

> ``omnitrace``), debugging tools (``rocgdb``), libraries and HIP with the runtime

Reviewer 2: ```suggestion
ROCm Compute Profiler), debugging tools (``rocgdb``), libraries and HIP with the runtime
```

Reviewer 2: Or maybe just remove omnitrace?

Reviewer 3: Removed, and renamed rocprof to rocprofv3

API and kernel language, to create heterogeneous applications running on both
CPUs and GPUs. ROCm provides marshalling libraries like
:doc:`hipFFT <hipfft:index>` or :doc:`hipBLAS <hipblas:index>` that act as a
thin programming layer over AMD ROCm and offer API compatiblity with CUDA
equivalent library. These libraries offer pointer-based memory interfaces and
are easily integrated into your applications.

HIP supports the ability to build and run on either AMD GPUs or NVIDIA GPUs.
GPU Programmers familiar with NVIDIA CUDA or OpenCL will find the HIP API
familiar and easy to use. You can quickly port your application to run on the

> available hardware while maintaining a single codebase. The :doc:`HIPify <hipify:index>`

Reviewer 2: This idea that HIP runs on NVIDIA GPUs was the thing I thought Saad was trying to eliminate. Check with him, but another approach is that HIP lets you quickly port CUDA code to run on AMD GPUs. This supports the "available hardware" theory, because your CUDA code runs on NVIDIA GPUs and your HIP code runs on AMD GPUs.
